### PR TITLE
Migrate cmd/ifqld  to platform

### PR DESCRIPTION
**Pull request by aanthony1243**
lots of file movement, minor code changes to resolve errors.  cmd/ifqld/main.go  still has many errors.

### CODE REVIEW

**Comment by jsternberg** [comment]
This looks good to me so far.

File: query/query_test/query_test_pkg.go~ [hunk 1]
```diff
@@ -0,0 +1 @@
+package query
```
Comment (nathanielc) [inline query/query_test/query_test_pkg.go~]: Can you remove this file?

File: query/query_test/query_test_pkg.go [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// no executable code in this directory, which contains tests for the query package
+package query
```
Comment (nathanielc) [inline query/query_test/query_test_pkg.go]: What is the purpose of this file?

Comment (aanthony1243) [reply query/query_test/query_test_pkg.go]: @desa   requested this change due to a build requirement that all directories contain at least one non-test go file.   An alternative would be to re-org the test dir so that it's one level up

Comment (jsternberg) [reply query/query_test/query_test_pkg.go]: I agree with moving this package, but I think we should delay anything that involves refactoring or reorganizing the directory structure until this is merged.

File: Gopkg.lock [hunk 3]
```diff
@@ -1,24 +1,115 @@
 # This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
 
 
+[[projects]]
```
Comment (nathanielc) [inline Gopkg.lock]: What is this a dep of?

File: Gopkg.lock [hunk 4]
```diff
@@ -1,24 +1,115 @@
...
+  name = "github.com/RoaringBitmap/roaring"
+  packages = ["."]
+  revision = "1a28a7fa985680f9f4e1644c0a857ec359a444b0"
+  version = "v0.4.7"
+
+[[projects]]
+  name = "github.com/Shopify/sarama"
```
Comment (nathanielc) [inline Gopkg.lock]: This doesn't look right. We should not have a dep on Kafka in this project.

File: Gopkg.lock [hunk 5]
```diff
@@ -1,24 +1,115 @@
...
   revision = "3a771d992973f24aa725d07868b467d1ddfceafb"
 
+[[projects]]
+  name = "github.com/cespare/xxhash"
+  packages = ["."]
+  revision = "5c37fe3735342a2e0d01c87a907579987c8936cc"
+  version = "v1.0.0"
+
+[[projects]]
+  name = "github.com/coreos/etcd"
```
Comment (nathanielc) [inline Gopkg.lock]: This also looks suspicious. We should not have a dep on etcd in this project.

File: cmd/ifqld/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# IFQL
+
+This process provides an implementation of the idpe.Query interface via the network.
+
+# Exposed Metrics
+
+The `ifqlde` process exposes a Prometheus endpoint on port `8093` by default.
```
Comment (nathanielc) [inline cmd/ifqld/README.md]: Update name here

File: cmd/ifqld/main.go [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+}
+
+var (
+	bindAddr         string
+	concurrencyQuota int
+	memoryBytesQuota int
+)
+
+func init() {
+	viper.SetEnvPrefix("IFQLDE")
```
Comment (nathanielc) [inline cmd/ifqld/main.go]: Update ifqlde -> ifqd throughout this file. Or should we make these changes in a separate commit?

File: cmd/ifqld/main.go [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+		//	Hosts:   hosts,
+		//	Timeout: etcdTimeout,
+		//}
+		//o := etcd.NewOrganizationCache(etcdConfig)
+		//err = o.Open()
+		//if err != nil {
+		//	logger.Error("failed to open etc organization cache", zap.Error(err))
+		//	os.Exit(1)
+		//}
+		//orgSvc = o
```
Comment (nathanielc) [inline cmd/ifqld/main.go]: Let's remove the bits about etcd here and just leave the first line of the comment.

File: cmd/ifqld/main.go [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+			},
+		}
+		hosts, err := bucketDiscover.Hosts()
+		logger.Info("connecting to bucket etcd hosts", zap.Strings("hosts", hosts))
+		if err != nil {
+			return errors.Wrap(err, "failed to discover bucket etcd hosts")
+		}
+
+		// TODO(adam): figure correct bucket service
+		//etcdConfig := etcd.Config{
```
Comment (nathanielc) [inline cmd/ifqld/main.go]: Same here

File: query/query_test/query_test_pkg.go (hunk 2 above)
Comment (nathanielc) [reply query/query_test/query_test_pkg.go]: yes, lets delay the change until after this is merged.

File: query/query_test/query_test_pkg.go~ (hunk 1 above)
Comment (nathanielc) [reply query/query_test/query_test_pkg.go~]: This still needs to be removed.

File: cmd/ifqld/main.go (hunk 7 above)
Comment (nathanielc) [reply cmd/ifqld/main.go]: I am fine if we make these rename changes in a follow up commit.

**Comment by nathanielc** [comment]
Also, lets make sure to rebase and squash this commit so its a clean commit without the intermediate commits with the extra deps etc.

File: cmd/ifqld/main.go (hunk 7 above)
Comment (aanthony1243) [reply cmd/ifqld/main.go]: only came up in these 2 places so I made the change.
